src/watchmaker/workers/salt.py

```diff
@@ -631,20 +632,22 @@ def process_states(self, states, exclude):
             return
 
         cmds = []
-        states = states.lower().split(',')
         salt_cmd = self.salt_state_args
 
-        if 'highstate' in states:
+        if 'highstate' in states.lower().split(','):
```

Comment:
The split/split/join/join is really hard to understand. I think we can do a little better. Here's one idea.

```suggestion
        highstate = False
        for state in states:
            if state.lower() == 'highstate':
                highstate = True
                states.remove(state)

        if highstate:
            self.log.info('Applying the salt "highstate"')
            cmds.append(salt_cmd + ['state.highstate'])
```

Reply:
Okay.  Let me give this suggestion a try.

Reply:
I updated to your suggestion.  Tests are passing as well.

Reply:
I think it reads well. Would still like to see a test case demonstrating case preservations for the "extra" states...